save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -91,34 +97,84 @@ class DownloadProjectController(private val configProperties: ConfigProperties)
         }
     }
 
+    @Suppress(
+        "LongMethod",
+        "ThrowsCount",
+        "TooGenericExceptionCaught",
+        "TOO_LONG_FUNCTION",
+        "LOCAL_VARIABLE_EARLY_DECLARATION")
     private fun sendToBackendAndOrchestrator(project: Project, path: String) {
         val execution = Execution(project, LocalDateTime.now(), LocalDateTime.now(),
             ExecutionStatus.PENDING, "1", path, 0, configProperties.executionLimit)
+        var execId: Long
         log.debug("Knock-Knock Backend")
         webClientBackend
             .post()
             .uri("/createExecution")
             .body(BodyInserters.fromValue(execution))
             .retrieve()
-            .onStatus({status -> status != HttpStatus.OK }) {
-                log.error("Backend internal error: ${it.statusCode()}")
+            .onStatus({status -> status != HttpStatus.OK }) { clientResponse ->
+                log.error("Backend internal error: ${clientResponse.statusCode()}")
                 throw ResponseStatusException(
                     HttpStatus.INTERNAL_SERVER_ERROR,
                     "Backend internal error"
                 )
             }
             .bodyToMono(Long::class.java)
             .doOnNext { executionId ->
-                // Post request to orchestrator to initiate its work
-                log.debug("Knock-Knock Orchestrator")
-                webClientOrchestrator
+                execId = executionId
+                webClientBackend
                     .post()
-                    .uri("/initializeAgents")
-                    .body(BodyInserters.fromValue(execution.also { it.id = executionId }))
+                    .uri("/saveTestSuites")
+                    .body(BodyInserters.fromValue(getAllTestSuites(project)))
                     .retrieve()
-                    .toEntity(HttpStatus::class.java)
-                    .subscribe()
+                    .onStatus({status -> status != HttpStatus.OK }) { clientResponse ->
+                        log.error("Backend internal error: ${clientResponse.statusCode()}")
+                        throw ResponseStatusException(
+                            HttpStatus.INTERNAL_SERVER_ERROR,
+                            "Backend internal error"
+                        )
+                    }
+                    .bodyToMono<List<TestSuite>>()
+                    .doOnNext { testSuiteList ->
```

Comment:
Please extract these callbacks into separate methods (e.g., returning `Mono`, so that you can call `doOnNext` on returned value)